boards/arduino/due/doc/index.rst

```diff
@@ -32,7 +32,7 @@ Hardware
 Supported Features
 ==================
 
-The arduino_due board configuration supports the following hardware features:
+The ``arduino_due`` board target supports the following hardware features:
```

Comment:
this is not a board target, it's the board name.

And the same goes for all other changes in this PR.

The cleanup is appreciated, but we should not swap one bad description for another.
I made a comment in the issue here: https://github.com/zephyrproject-rtos/zephyr/issues/79621#issuecomment-2490674540

Feel free to decide whether to use:
- Full name, in this case: `Arduino Due`
- Board name: in this case: `arduino_due`
- Board target, in this case: `arduino_due/sam3x8e`


I think in this context using the full name, `Arduino Due` looks nicest, but overall using the board name makes most sense because board name is defined for all boards. (Not all boards defines a `Full name` in the board.yml).
And I believe the text in general tries to refer to the high level board name, and not a specific board target.

If you decide to choose board target, then please ensure that a valid / correct board target is used in the text.